File: tests/test_code_from_python.cs
using System;
using System.Math;  // from python: math
using System.Random;  // from python: random
using time;
using System.Environment;  // from python: os
using sys;

namespace testcodeoriginal
{
    public static class Functions
    {
        public static void Clear()
        {
            os.System(((os.Name == "nt") ? "cls" : "clear"));
        }

        public static string Galaxy(double width = 120, double height = 40, int t = 0.0d, int arms = 3)
        {
            object output = new[] {  };
            object  = <unknown>;
            foreach (var y in range(height))
            {
                string row = "";
                foreach (var x in range(width))
                {
                    object  = <unknown>;
                    object r = math.Sqrt((Math.Pow(dx, 2) + Math.Pow(dy, 2)));
                    object a = math.Atan2(dy, dx);
                    int swirl = (((a * arms) + (r * 12)) - (t * 2));
                    double noise = ((pnoise2((dx * 2), ((dy * 2) + t)) * 0.5d) + 0.5d);
                    object bright = Math.Pow((math.Cos(swirl) * noise), 2);
                    if ((bright > (0.5d - (r * 0.5d))))
                    {
                        object color = cOLORS[(int(((bright + (random.Random() * 0.1d)) * (len(cOLORS) - 1))) % len(cOLORS))];
                        object char = random.Choice(new[] { "*", "·", "✦", ".", "•" });
                        row = (row + null);
                    }
                    else
                    {
                        row = (row + " ");
                    }
                }
                output.Append(row);
            }
            return "
".Join(output);
        }

        public static void Animate(object frames = 99999)
        {
            int t = 0;
            try
            {
                while (true)
                {
                    clear();
                    print(galaxy(120, 40, t));
                    print(null);
                    t = (t + 0.1d);
                    time.Sleep(0.08d);
                }
            }
            catch (KeyboardInterrupt)
            {
                clear();
                print("🌀 Galaxy collapsed. Goodbye.
");
            }
        }

    }

}

File: tests/test_code_from_python.rs
use math;
use random;
use time;
use os;
use sys;

pub fn clear() {
    os.system(if (os.name == "nt") { "cls" } else { "clear" });
}

pub fn galaxy(width: f64, height: f64, t: i32, arms: i32) -> String {
    let output: Box<dyn std::any::Any> = vec![];
    let : Box<dyn std::any::Any> = <unknown>;
    for y in range(height) {
        let row: String = "";
        for x in range(width) {
            let : Box<dyn std::any::Any> = <unknown>;
            let r: Box<dyn std::any::Any> = math.sqrt(((dx ** 2) + (dy ** 2)));
            let a: Box<dyn std::any::Any> = math.atan2(dy, dx);
            let swirl: i32 = (((a * arms) + (r * 12)) - (t * 2));
            let noise: f64 = ((pnoise2((dx * 2), ((dy * 2) + t)) * 0.5) + 0.5);
            let bright: Box<dyn std::any::Any> = ((math.cos(swirl) * noise) ** 2);
            if (bright > (0.5 - (r * 0.5))) {
                let color: Box<dyn std::any::Any> = colors[(int(((bright + (random.random() * 0.1)) * (len(colors) - 1))) % len(colors))];
                let char: Box<dyn std::any::Any> = random.choice(vec!["*", "·", "✦", ".", "•"]);
                row = (row + None);
            } else {
                row = (row + " ");
            }
        }
        output.append(row);
    }
    return "
".join(output);
}

pub fn animate(frames: &Box<dyn std::any::Any>) {
    let t: i32 = 0;
    // try-catch block
    while true {
        clear();
        print(galaxy(120, 40, t));
        print(None);
        t = (t + 0.1);
        time.sleep(0.08);
    }
    // catch KeyboardInterrupt
    clear();
    print("🌀 Galaxy collapsed. Goodbye.
");
}
